game.py

```diff
@@ -1,62 +1,131 @@
+            return 10
+        elif rank in ACE:
+            return (1, 11)
+        else:
+            return rank
+
     def __str__(self):
-        return f"{self.value}{self.suit}"
+        return f"{self.rank}{self.suit}"
 
     def __repr__(self):
-        return f"Card('{self.value}', '{self.suit}')"
+        return f"Card('{self.rank}', '{self.suit}')"
 
 
 class Deck:
 
-    CARD_VALUE = ["2", "3", "4", "5", "6", "7", "8", "9", "10", "J", "Q", "K", "A"]
-    CARD_SUIT = ["\u2663", "\u2665", "\u2666", "\u2660"]
+    CARD_SUITS = ["♠️", "♣️", "♥️", "♦️"]
 
     def __init__(self):
-        self.deck = []
-        self.new_deck()
+        self.cards = [
+            Card(rank, suit) for suit in self.CARD_SUITS for rank in range(1, 14)
+        ]
+
+        self.cards = self.shuffle_deck(self.cards)
 
-    def new_deck(self):
-        for suit in self.CARD_SUIT:
-            for value in self.CARD_VALUE:
-                self.deck.append(Card(value, suit))
-        random.shuffle(self.deck)
+    def shuffle_deck(self, cards):
+        return shuffle(cards)
+
+    def deal_card(self):
+        return self.cards.pop(0)
 
     def __str__(self):
-        return f"{[str(card) for card in self.deck]}"
+        return f"{[str(card) for card in self.cards]}"
 
 
 class Hand:
     def __init__(self):
         self.cards = []
 
-    # Todo
-    # should be able to add cards to the hand
-    # should be able to display / show all the cards in the hand and total value
+    def add_card(self, card):
+        self.cards.append(card)
+
+    def show_cards(
+        self,
+    ):  # from a testing perspective should I use the class attribute or paramter
+        return [card.__str__() for card in self.cards]
+
+    def show_first_card(self):
+        return self.cards[0]
+
+    def show_hand_value(self):
+        rank = 0
+        hand_value = 0
+        for card in self.cards:
+            rank = card.check_card_value(card)
+            if type(rank) is tuple:
+                rank = 1
+            else:
+                rank = card.rank
+            hand_value += rank
+        return hand_value
+
+        # return sum([ card.check_card_value(card) for card in self.cards])
+
+    def is_blackjack(self):
+        if self.show_hand_value() == 21:
+            return True
+        else:
+            return False
+
+    def is_bust(self):
+        if self.show_hand_value() > 21:
+            return True
+        else:
+            return False
 
 
 class Player:
     def __init__(self):
         self.hand = Hand()
 
-    # Todo
-    # player should be able to request a hit or hold
+    def request_hit(self, card):
+        self.hand.add_card(card)
+
+    def stand(self):
+        pass
+
+
+class Dealer(Player):
+    def __init__(self):
+        Player.__init__(self)
+
+
+class Game:
+    def __init__(self):
+        self.player = Player()
+        self.dealer = Dealer()
+        self.deck = Deck()
+
+    def play_round(self):
+        deck = Deck()
```

Comment:
you make an instance every round, would it be more realistic to use the same deck throughout the game?